### PR TITLE
docs: Add autoscale recommendations docs

### PR DESCRIPTION
**Pull request by spikecurtis**
We get asked about this enough, e.g. https://github.com/coder/v2-customers/issues/173 that we should have a place in the docs we can just point people to.

### CODE REVIEW

File: docs/admin/scale.md [hunk 1]
```diff
@@ -64,6 +64,33 @@ The test does the following:
...
+connections will drop. An even larger proportion of users could be affected if they use applications that use more
+than one websocket.
+
+The severity of the interruption varies by application. Coder's web terminal, for example, will reconnect to the same
+session and continue. So, this should not be interpreted as saying coderd replicas should never be taken down for any
+reason.
+
+We recommend you plan to run enough coderd replicas to comfortably meet your weekly high-water-mark load, and monitor
+coderd peak CPU & memory utilization over the long term, reevaluating periodically. When scaling down (or performing
+upgrades), schedule these on off hours to minimize user interruptions.
```
Comment (johnstcn) [inline docs/admin/scale.md:85]: nit: "on off hours" reads funny, suggest

```suggestion
upgrades), schedule these outside normal working hours to minimize user interruptions.
```

File: docs/admin/scale.md [hunk 2]
```diff
@@ -64,6 +64,33 @@ The test does the following:
...
+
+### A note for Kubernetes users
+
+When running on Kubernetes on cloud infrastructure (i.e. not bare metal), many operators choose to employ a _cluster_
+autoscaler that adds and removes Kubernetes _nodes_ according to load. Coder can coexist with such cluster autoscalers,
+but we recommend you take steps to prevent the autoscaler from evicting coderd pods, as an eviction will cause the same
+interruptions as described above.
```
Comment (johnstcn) [inline docs/admin/scale.md:92]: Might be no harm to add an example:

```suggestion
interruptions as described above. For example, if you are using the [Kubernetes cluster
autoscaler](https://kubernetes.io/docs/reference/labels-annotations-taints/#cluster-autoscaler-kubernetes-io-safe-to-evict), you may wish to set
`cluster-autoscaler.kubernetes.io/safe-to-evict: "false"` as an annotation
on the coderd deployment.
```

**Review by johnstcn — APPROVED** [review]
Nice! Just a couple of suggestions but otherwise 👍

**Review by bpmct — APPROVED** [review]

**Comment by bpmct** [comment]
This is really helpful, thanks for doing this!

**Pull request merged**